core/src/main/java/org/radargun/stages/test/legacy/LegacyStressor.java

```diff
@@ -87,15 +91,18 @@ private void runInternal() {
                if (ongoingTx != null) {
                   endTransactionAndRegisterStats(null);
                }
-               stats.begin();
-               this.started = started;
                break;
-            }
-            Operation operation = operationSelector.next(random);
-            try {
-               logic.run(operation);
-            } catch (OperationLogic.RequestException e) {
-               // the exception was already logged in makeRequest
+            } else {
```

Comment:
That is exactly what the code does to guarantee that every thread spawned from the stage is ready to run before starting to collect statistics. I thought this was what we discussed?